### PR TITLE
Updated the docker deployment for two services and update the README.md file with setup guide.

### PR DESCRIPTION
**Pull request by yashodgayashan**
### Overview
- Added scripts to use in docker environment and the local environment.
- Update the example.conf with new configurations.
- Update the README.md file with setup guide.

Fixes: https://github.com/yashodgayashan/environment_platform/issues/26

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
-# environment_platform
\ No newline at end of file
+# environment_platform
+
+## Core
```
Comment (deshankoswatte) [inline README.md:3]:
```suggestion
## Module Core(contains the back-end functionalities).
```

File: README.md [hunk 2]
```diff
@@ -1 +1,56 @@
...
+
+The following instruction must be followed to setup the backend in your local machine
+
+### Requirments
```
Comment (deshankoswatte) [inline README.md:7]:
```suggestion
### Prerequisites
```

File: README.md [hunk 3]
```diff
@@ -1 +1,56 @@
...
+
+- Ballerina 1.2
+- Docker
+
+### Build
```
Comment (deshankoswatte) [inline README.md:12]:
```suggestion
### How to Build?
```

File: README.md [hunk 4]
```diff
@@ -1 +1,56 @@
...
+
+To build the Core please execute the following command
+
+```
+./setup.sh
+```
+
+### Configuration file
+
+Please Make a file called `ballerina.conf` in core/resources as given in the `example.conf` and update the values.
```
Comment (deshankoswatte) [inline README.md:22]:
```suggestion
Make a file named `ballerina.conf` in `core/resources`. You can refer the file `example.conf` located at `core/resources/` for more details. Then, update the created file with your preference of values.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,56 @@
...
+
+### Build
+
+To build the Core please execute the following command
+
+```
+./setup.sh
+```
+
+### Configuration file
```
Comment (deshankoswatte) [inline README.md:20]:
```suggestion
### Create a configuration file
```

File: README.md [hunk 6]
```diff
@@ -1 +1,56 @@
...
+```
+
+##### Terminate the docker container
+
+- First press `ctl+c`.
+- Then run `docker-compose down`
+
+#### Develop and run using docker
+
+When you develop a feature and run using containers. You have to build the jar files using `./setup.sh` and then you have to build and up the dokcer composer.
```
Comment (deshankoswatte) [inline README.md:47]:
```suggestion
When you develop a feature and run using containers. You have to build the jar files using `./setup.sh` and then you have to build and up the docker composer.
```

File: README.md [hunk 7]
```diff
@@ -1 +1,56 @@
...
+
+# Run the project using docker
+
+**Important** - Make sure that configuration files are setup for local machine.
```
Comment (deshankoswatte) [inline README.md:51]:
```suggestion
**Important** - Make sure that the configuration files are setup for your local machine.
```

File: README.md [hunk 8]
```diff
@@ -1 +1,56 @@
...
+
+To run the project,
+
+- First `cd core`.
```
Comment (deshankoswatte) [inline README.md:55]:
```suggestion
- Execute the command `cd core` to navigate to the `core` module.
```

File: README.md [hunk 9]
```diff
@@ -1 +1,56 @@
...
+- Then run `./scripts/run-local.sh`.
```
Comment (deshankoswatte) [inline README.md:56]:
```suggestion
- Then, run the script using the command `./scripts/run-local.sh`.
```

File: core/resources/example.conf [hunk 10]
```diff
@@ -1,11 +1,25 @@
...
+# # docker deployment
+# KEY_STORE_PATH="/env_platform/resources/ballerinaKeystore.p12"
+
 JWT_EXPIRE=3600 # Expires in one hour
 
 TRUST_STORE_PASSWORD = <password>
+# development locally 
+TRUST_STORE_PATH="resources/ballerinaTruststore.p12"
+# # docker deployment
+# TRUST_STORE_PATH="/env_platform/resources/ballerinaTruststore.p12"
```
Comment (deshankoswatte) [inline core/resources/example.conf:25]:
```suggestion
# TRUST_STORE_PATH="/env_platform/resources/ballerinaTruststore.p12"

```

File: core/scripts/run-docker.sh [hunk 11]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+ballerina run ./target/env_platform.jar --b7a.config.file=./resources/ballerina.conf &
+ballerina run ./target/auth_service.jar --b7a.config.file=./resources/ballerina.conf
```
Comment (deshankoswatte) [inline core/scripts/run-docker.sh:4]:
```suggestion
ballerina run ./target/auth_service.jar --b7a.config.file=./resources/ballerina.conf 

```

File: core/scripts/run-local.sh [hunk 12]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+ballerina run target/bin/env_platform.jar --b7a.config.file=resources/ballerina.conf &
+ballerina run target/bin/auth_service.jar --b7a.config.file=resources/ballerina.conf
```
Comment (deshankoswatte) [inline core/scripts/run-local.sh:4]:
```suggestion
ballerina run target/bin/auth_service.jar --b7a.config.file=resources/ballerina.conf 

```

**Comment by yashodgayashan** [comment]
@deshankoswatte Resolved requested changes.

File: README.md [hunk 13]
```diff
@@ -1 +1,56 @@
...
+
+##### Run the image
+
+```
+docker-compose up
+```
+
+##### Terminate the docker container
+
+- First press `ctl+c`.
```
Comment (deshankoswatte) [inline README.md:42]:
```suggestion
- Press `Ctrl+C` to terminate the program.
```

File: README.md [hunk 14]
```diff
@@ -1 +1,56 @@
...
+- Then run `docker-compose down`
```
Comment (deshankoswatte) [inline README.md:43]:
```suggestion
- Run `docker-compose down` to stop your containers.
```

**Comment by yashodgayashan** [comment]
> @yashodgayashan, @sanjulamadurapperuma, @vikumkbv I think is good enough for now but certainly we should improve this a lot in the future.

Yes I just make the configurations because need to added authentication and authorisation to the platform.

File: README.md [hunk 15]
```diff
@@ -1 +1,56 @@
...
+##### Terminate the docker container
+
+- Press `Ctrl+C` to terminate the program.
+- Run `docker-compose down` to stop your containers.
+
+#### Develop and run using docker
+
+When you develop a feature and run using containers. You have to build the jar files using `./setup.sh` and then you have to build and up the docker composer.
+
+# Run the project using docker
```
Comment (deshankoswatte) [inline README.md:49]: Isn't this run the project locally? And if it is:
```suggestion
### Run the project locally
```

File: README.md [hunk 16]
```diff
@@ -1 +1,56 @@
...
+
+**Important** - Make sure that the configuration files are setup for your local machine.
+
+To run the project,
```
Comment (deshankoswatte) [inline README.md:53]:
```suggestion
To run the module,
```

File: README.md [hunk 17]
```diff
@@ -1 +1,56 @@
...
+
+```
+./setup.sh
+```
+
+### Create a configuration file
+
+Make a file named `ballerina.conf` in `core/resources`. You can refer the file `example.conf` located at `core/resources/` for more details. Then, update the created file with your preference of values.
+
+### Run the project using docker
```
Comment (deshankoswatte) [inline README.md:24]:
```suggestion
### Run the module using docker
```

File: README.md [hunk 18]
```diff
@@ -1 +1,56 @@
-# environment_platform
\ No newline at end of file
+# environment_platform
```
Comment (deshankoswatte) [inline README.md:1]:
```suggestion
# Environment Platform
```

**Comment by deshankoswatte** [comment]
@sanjulamadurapperuma please review and approve.

**Pull request merged**